### PR TITLE
Rename retree into satellite (fixes #1451)

### LINKED ISSUE #1451: Retree: Rename retree to satellite
At the end of last year [a new exercise was added](https://github.com/exercism/problem-specifications/pull/1424) called "retree". I would like us to get into the habit of naming exercises by what the story is about, not what they teach. That makes the exercise more appealing in the UI (think `bob` vs `trinary` or `food-chain` vs `transpose`). I feel like this exercise is about transmitting data to Alpha Centauri - so I'd like to propose renaming it to "Satellite" instead. As it's a new exercise, I don't think this will be a big problem.

What are people's thoughts? (cc @fidelcoria @sshine @ErikSchierboom @rpottsoh @nicolechalmers @kytrinyx)

### PR DESCRIPTION
**Pull request by sshine**
Following @petertseng's suggested workflow in #1451 (thanks for providing this, by the way!):

- [x] Actually perform the rename
- [x] ~~Send a PR to https://github.com/exercism/website-icons/tree/master/exercises~~ The icon for "retree" has not been merged yet, so instead, ping exercism/website-icons#22 and have them name this icon ~~"sattelite"~~ "satellite" instead of "retree".
- [x] @cmccandless (Python) and @fidelcoria (Prolog) both agreed to rename the exercise locally: You can go ahead and rename the exercise in the `exercises` directory and in `config.json`!
  - [x] It's been renamed in Python
  - [ ] It's been renamed in Prolog
- [x] ~~Rename mentor notes~~ No mentor notes for exercise yet.

This PR is mergeable before these checkboxes have been marked.

**Question:** Should the test version be bumped when the exercise name changes?

### CODE REVIEW

File: exercises/sattelite/canonical-data.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "exercise": "retree",
+  "exercise": "sattelite",
```
Comment (cmccandless) [inline exercises/sattelite/canonical-data.json]:
```suggestion
  "exercise": "satelite",
```

File: exercises/sattelite/metadata.yml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 ---
-title: "Retree"
+title: "Sattelite"
```
Comment (cmccandless) [inline exercises/sattelite/metadata.yml]:
```suggestion
title: "Satelite"
```

File: exercises/sattelite/canonical-data.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
   "version": "1.0.0",
```
Comment (cmccandless) [inline exercises/sattelite/canonical-data.json]: Should we modify the version here?

Comment (sshine) [reply exercises/sattelite/canonical-data.json]: I think so.

> MAJOR changes should be expected to break even well-behaved test generators.

**Comment by sshine** [comment]
Darn. That's how you spell it in Danish!

I've fixed the typo everywhere but in the name of my feature branch.

(Not sure how to do that without abandoning the PR.)

**Comment by cmccandless** [comment]
> I've fixed the typo everywhere but in the name of my feature branch.

I wouldn't worry about it.

> (Not sure how to do that without abandoning the PR.)

I'm not sure it's possible.

**Pull request merged**